packages/peer-exchange/src/waku_peer_exchange.ts

```diff
@@ -71,12 +66,17 @@ export class WakuPeerExchange implements PeerExchange {
         throw new Error("Failed to decode response");
       }
 
-      const peerInfos: PeerInfo[] = [];
+      const enrPromises: Promise<ENR>[] = [];
+      for (const peerInfo of decoded.peerInfos) {
+        if (!peerInfo.ENR) continue;
+        const enr = ENR.decode(peerInfo.ENR);
+        enrPromises.push(enr);
+      }
 
-      decoded.peerInfos.forEach(async (peerInfo) => {
-        if (!peerInfo.ENR) return;
-        const enr = await ENR.decode(peerInfo.ENR);
-        peerInfos.push({ ENR: enr });
+      const peerInfos = (await Promise.all(enrPromises)).map((enr) => {
+        return {
+          ENR: enr,
+        };
       });
 
       return { peerInfos };
```

Comment:
You currently have both of worst world:

1. You have two iterations: `for of` and `map`
2. You are using `map` on the result of `await Promise.all` forcing you to wrap it in brackets.

Either make it readable (ie, solve (2)) or compact (solve (1)).

Reply:
addressed - lmk if looks good :)